pool/_RPC_dry.gno

```diff
@@ -58,9 +58,12 @@ func DrySwap(
 		feeGrowthGlobalX128 = pool.feeGrowthGlobal1X128
 	}
 
-	pool.slot0.unlocked = false
+	// pool.slot0.unlocked = false
+	slot0 := pool.slot0
+	slot0.unlocked = false
+
 	cache := newSwapCache(feeProtocol, pool.liquidity)
-	state := pool.newSwapState(amountSpecified, feeGrowthGlobalX128, cache.liquidityStart) // TODO: feeGrowthGlobalX128.Clone() or NOT
+	state := newSwapState(amountSpecified, feeGrowthGlobalX128, cache.liquidityStart, slot0) // TODO: feeGrowthGlobalX128.Clone() or NOT
```

Comment:
I think you should do `.Clone()`. Passing pointer value can lead many issues.
@onlyhyde WDYT

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/403/commits/92d7e60ed3167ede13cf8539d6d23631bb03dc5c